src/attached_probe.cpp

```diff
@@ -1361,7 +1362,7 @@ void AttachedProbe::attach_profile()
                                               group_fd);
 
     if (perf_event_fd < 0)
-      LOG(FATAL) << "Error attaching probe: " << probe_.name;
+      LOG(BUG) << "Error attaching probe: " << probe_.name;
```

Comment:
and here (and for rest of file)

Reply:
So should all of these be `LOG(WARNING)`?

Reply:
Should probably throw runtime_error, I think. We cannot attach a subset of user requested probes. And I can't think of any way to recover higher up in the stack. If recovery is possible locally, it can be done before exception is thrown.

Reply:
I think a `LOG(WARNING)` coupled with an early return is the same thing here as (according to our code guidelines) we don't want to throw recoverable errors, which this is.

Reply:
How come these are recoverable? When probe attachment fails, bpftrace should also fail so why is this not throwing `FatalUserException`?

Reply:
@viktormalik History is on your side in that before I started changing the logging we used to throw a `std::runtime_error` and failing hard in [bpftrace.cpp](https://github.com/bpftrace/bpftrace/blob/4c8b262aed6399a04126d2f5199d0d588924e6d2/src/bpftrace.cpp#L946-L947). I can change these to `FatalUserException` (though I wish we had more info on the cause for the users). Alternatively, if even one probe is successful bpftrace can proceed but if there is user logic where two probes depend on one another than continuing would be worthless.

Reply:
Ok, switched these all back to throw. Sorry for the churn.